packages/clerk-js/src/ui/userButton/useUserButtonVisibility.ts

```diff
@@ -0,0 +1,15 @@
+import { useDetectClickOutside } from '@clerk/shared/hooks';
+import React from 'react';
+import { useCoreClerk } from 'ui/contexts';
+
+export function useUserButtonVisibility(userButtonContainer: React.RefObject<HTMLDivElement>) {
+  const clerk = useCoreClerk();
+
+  const { isActive, setIsActive } = useDetectClickOutside(userButtonContainer, false);
+
+  /** Allow programmatic access of the UserButton visibility state */
+  clerk.openUserButton = () => setIsActive(true);
```

Comment:
@igneel64 This is a direct mutation on the clerk singleton which is a bit unexpected. Is there a way to toggle the UserButton open/close state via the Components renderer?

Moreover, can the `clerk-react` API support an `open` prop: `<UserButton open />` ?